staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -307,72 +243,30 @@ func TestSyncLoadBalancerIfNeeded(t *testing.T) {
 		},
 		{
 			desc: "service that needs cleanup",
-			service: &v1.Service{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "basic-service1",
-					Namespace: "default",
-					DeletionTimestamp: &metav1.Time{
-						Time: time.Now(),
-					},
-					Finalizers: []string{servicehelper.LoadBalancerCleanupFinalizer},
-				},
-				Spec: v1.ServiceSpec{
-					Ports: []v1.ServicePort{{
-						Port:     80,
-						Protocol: v1.ProtocolTCP,
-					}},
-					Type: v1.ServiceTypeLoadBalancer,
-				},
-				Status: v1.ServiceStatus{
-					LoadBalancer: v1.LoadBalancerStatus{
-						Ingress: []v1.LoadBalancerIngress{
-							{IP: "8.8.8.8"},
-						},
-					},
-				},
-			},
+			service: newService(
+				"basic-service1", "", v1.ServiceTypeLoadBalancer,
+				tweakAddFinalizers([]string{servicehelper.LoadBalancerCleanupFinalizer}),
+				tweakAddPorts(v1.ProtocolTCP, ""),
+				tweakAddLBI("8.8.8.8"),
+				tweakAddDeletionTimestamp(time.Now()),
+			),
 			lbExists:             true,
 			expectOp:             deleteLoadBalancer,
 			expectDeleteAttempt:  true,
 			expectPatchStatus:    true,
 			expectPatchFinalizer: true,
 		},
 		{
-			desc: "service without finalizer that wants LB",
-			service: &v1.Service{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "basic-service1",
-					Namespace: "default",
-				},
-				Spec: v1.ServiceSpec{
-					Ports: []v1.ServicePort{{
-						Port:     80,
-						Protocol: v1.ProtocolTCP,
-					}},
-					Type: v1.ServiceTypeLoadBalancer,
-				},
-			},
+			desc:                 "service without finalizer that wants LB",
+			service:              newService("basic-service1", "", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, "")),
```

Comment:
Isn't this the same as a case above?

Reply:
yeah, it's identical with above "tcp service that wants LB"